### DISCUSSION TITLE
lib/arm64-v8a/libcrypto.so

### DISCUSSION THREAD

**Discussion by farzadkabiri-nrp** [ThibaultBee/StreamPack#151] [General]
Hi
i'm using srtdroid-core & srtdroid-ktx & streampack & streampack-extension-srt in my POC (push to talk over cellular) app, after updated to this new versions i got this error: 
2 files found with path 'lib/arm64-v8a/libcrypto.so' from inputs:
 - C:\Users\Kabiri\.gradle\caches\transforms-4\2d45959302e794e13c6f39ee61f48f81\transformed\jetified-srtdroid-core-1.8.0\jni\arm64-v8a\libcrypto.so
 - C:\Users\Kabiri\.gradle\caches\transforms-4\d4b770380117cc5c7047cd7b9c71bb41\transformed\jetified-srtdroid-1.7.0\jni\arm64-v8a\libcrypto.so
 
and next i used 
    packagingOptions {
        pickFirst("lib/x86/libcrypto.so")
        pickFirst("lib/x86_64/libcrypto.so")
       pickFirst("lib/armeabi-v7a/libcrypto.so")
        pickFirst("lib/arm64-v8a/libcrypto.so")
    }
bot the error is remain.
 how can resolve that.

**Comment by ThibaultBee** [comment]
Hi,
Have you tried the example in the README.md:

```
android {
    packagingOptions {
        pickFirst '**/*.so'
    }
}
```

**Reply by farzadkabiri-nrp** [reply]
yes i do and i got this error:

class com.android.build.gradle.internal.dsl.PackagingOptions$AgpDecorated_Decorated cannot be cast to class com.android.build.api.dsl.PackagingOptions (com.android.build.gradle.internal.dsl.PackagingOptions$AgpDecorated_Decorated is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader$InstrumentingVisitableURLClassLoader @697b05c3; com.android.build.api.dsl.PackagingOptions is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader$InstrumentingVisitableURLClassLoader @26e096f5)

**Reply by ThibaultBee** [reply]
Are you using Kotlin DSL?
`packagingOptions` might have changed in the last version.

**Reply by farzadkabiri-nrp** [reply]
thanks  for this library. it is amazing
yes, i use kotlin dsl,

**Comment by ThibaultBee** [comment]
Have you tried a Kotlin DSL format like this: (not tested)
```
android {
    packagingOptions {
        pickFirst("**/*.so")
    }
}
```

**Reply by farzadkabiri-nrp** [reply]
I also used this and invalidate caches, but unfortunately it gives the same error again:

class com.android.build.gradle.internal.dsl.PackagingOptions$AgpDecorated_Decorated cannot be cast to class com.android.build.api.dsl.PackagingOptions (com.android.build.gradle.internal.dsl.PackagingOptions$AgpDecorated_Decorated is in unnamed module of loader

**Reply by ThibaultBee** [reply]
I guess something with Kotlin DSL has changed. Have you look on stackoverflow?
In which build.gradle.kts, have you put these lines?

Also, as StreamPack 2.6.1 uses srtdroid 1.7.0, you might not have the issue if you also uses srtdroid 1.7.0.

**Reply by farzadkabiri-nrp** [reply]
yes i look on stackoverflow, i do in app build.gradle.kts's android block, 
No problem using srtdroid = 1.7.0 and streampack = 2.6.1. This problem occurs when I update to version srtdroid = 1.8.0 and streampack = 2.6.1

**Comment by ThibaultBee** [comment]
Will be resolved with 3.1.0
